src/main/java/com/internet/shop/dao/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO products(name, price) VALUES(?, ?)";
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, item.getName());
+            preparedStatement.setDouble(2, item.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new RuntimeException("Product " + item + " was not created", e);
```

Comment:
Let’s create custom exception DataProcessingException with two parameters: String message and Throwable ex.

Reply:
added